rust/lancedb/src/remote/table.rs

```diff
@@ -325,6 +325,9 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
             message: "table_definition is not supported on LanceDB cloud.".into(),
         })
     }
+    fn dataset_uri(&self) -> &str {
+        "NOT_SUPPORTED"
```

Comment:
throwing an exception could be better?

Reply:
Why can't we support it?  We can just latch the `uri` when the table is constructed.